### PR TITLE
Reformat installation docs to be more readable

### PR DESCRIPTION
**Pull request by liu-cong**
Fixes #

## Proposed Changes
- Just reformatting. No additional updates.
- Existing docs have too many levels of indentation, which makes them very hard to follow
- Also moved manual configuration steps for auth to a separate doc so users don't have to read the details if they are not interested.
- 

**Release Note**

```release-note
NONE
```

**Docs**

cc @grac3gao

### CODE REVIEW

File: docs/install/install-knative-gcp.md [hunk 1]
```diff
@@ -14,180 +16,67 @@
...
+
+### Option 1: Install from Master using [ko](http://github.com/google/ko)
+
+```shell
+ko apply -f ./config
+```
+
+### Option 2: Install a [release](https://github.com/google/knative-gcp/releases).
+
+Remember to update `{{< version >}}` in the commands below with the
```
Comment (Harwayne) [inline docs/install/install-knative-gcp.md:29]: Can we change version to a shell variable? So that users set it once and then they can copy-paste the other commands.

File: docs/install/install-knative-gcp.md [hunk 2]
```diff
@@ -14,180 +16,67 @@
...
+`gcloud config set compute/zone $ZONE` to set it.
 
-           ```shell
-           gcloud services enable iamcredentials.googleapis.com
-           gcloud beta container clusters update $CLUSTER_NAME \
-           --identity-namespace=$PROJECT_ID.svc.id.goog
-           ```
+**_Note_**: Both scripts will have a step to create a Google Cloud Service
+Account `cloud-run-events`. Ignore the error message if you already had
+this service account (error for 'service account already exists').
```
Comment (Harwayne) [inline docs/install/install-knative-gcp.md:64]: Is it worth altering the scripts so that users don't see an error? Not in this PR, but maybe as a backlog task (with a TODO here).

File: docs/install/pubsub-service-account.md [hunk 3]
```diff
@@ -6,137 +6,137 @@ data plane component, which basically needs permissions to read and/or write to
...
-        encompasses both of the above plus some other permissions. Note that if
-        you prefer finer-grained privileges, you can just grant the ones
-        mentioned above.
+It is the recommended way to access Google Cloud services from within
+GKE due to its improved security properties and manageability. For more
+information about Workload Identity see
+[here](https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity).
+
+1. Enable Workload Identity. Skip this step if you already enable it in
+  the control plane setup: [Install Knative-GCP](install-knative-gcp.md).
```
Comment (Harwayne) [inline docs/install/pubsub-service-account.md:63]:
```suggestion
1. Enable Workload Identity. Skip this step if you have already enabled it
  during control plane setup: [Install Knative-GCP](install-knative-gcp.md).
```

File: docs/install/pubsub-service-account.md [hunk 4]
```diff
@@ -6,137 +6,137 @@ data plane component, which basically needs permissions to read and/or write to
...
+    ```shell
+    gcloud services enable iamcredentials.googleapis.com
+    gcloud beta container clusters update ${CLUSTER_NAME} \
+    --identity-namespace=${PROJECT_ID}.svc.id.goog
+    ```
+
+1. Give `iam.serviceAccountAdmin` role to your control plane's Google
+  Cloud Service Account. You can skip this step if you grant
+  `roles/owner` privileges in the control plane setup:
+  [Install Knative-GCP](install-knative-gcp.md).
```
Comment (Harwayne) [inline docs/install/pubsub-service-account.md:74]:
```suggestion
1. Give `iam.serviceAccountAdmin` role to your control plane's Google
  Cloud Service Account. You can skip this step if you granted
  `roles/owner` privileges during control plane setup:
  [Install Knative-GCP](install-knative-gcp.md).
```

File: docs/install/pubsub-service-account.md [hunk 5]
```diff
@@ -6,137 +6,137 @@ data plane component, which basically needs permissions to read and/or write to
...
+  you just created when creating resources. Check
+  [example](https://github.com/google/knative-gcp/tree/master/docs/examples)
+  of each resource for detail.
+
+    Generally, updating `spec.googleServiceAccount` will trigger the
+    controller to enable Workload Identity for sources in the data plane.
+    The following steps are handled by the controller:
+
+    1. Create a Kubernetes Service Account and add an `OwnerReference` to
+     it. The `OwnerReference` is referencing to the source.
```
Comment (Harwayne) [inline docs/install/pubsub-service-account.md:93]:
```suggestion
    1. Create a Kubernetes Service Account with an `OwnerReference` of
     the source.
```

File: docs/install/pubsub-service-account.md [hunk 6]
```diff
@@ -6,137 +6,137 @@ data plane component, which basically needs permissions to read and/or write to
...
+
+    1. Bind the Kubernetes Service Account with Google Cloud Service
+     Account, this will add `role/iam.workloadIdentityUser` to the
+     Google Cloud Service Account. The scope of this role is only for
+     this specific Google Cloud Service Account. It equals to this
```
Comment (Harwayne) [inline docs/install/pubsub-service-account.md:98]:
```suggestion
     this specific Google Cloud Service Account. It is equivalent to this
```

File: docs/install/pubsub-service-account.md [hunk 7]
```diff
@@ -6,137 +6,137 @@ data plane component, which basically needs permissions to read and/or write to
...
-            kubectl --namespace default create secret generic google-cloud-key --from-file=key.json=cre-pubsub.json
-            ```
-
-            `google-cloud-key` and `key.json` are default values expected by our
-            resources.
+    1. Annotate the Kubernetes Service Account with
+     `iam.gke.io/gcp-service-account=GCP-service-account@PROJECT_ID.iam.gserviceaccount.com`
+
+    For every namespace, the Kubernetes Service Account and the Google
+    Cloud Service Account will have one to one mapping relation. If
```
Comment (Harwayne) [inline docs/install/pubsub-service-account.md:112]:
```suggestion
    Cloud Service Account will have a one to one mapping. If
```

File: docs/install/pubsub-service-account.md [hunk 8]
```diff
@@ -6,137 +6,137 @@ data plane component, which basically needs permissions to read and/or write to
...
+    multiple source instances use the same Google Cloud Service Account,
+    they will use the same Kubernetes Service Account as well, and the
+    kubernetes Service Account will have multiple `OwnerReference`s. If
```
Comment (Harwayne) [inline docs/install/pubsub-service-account.md:115]:
```suggestion
    Kubernetes Service Account will have multiple `OwnerReference`s. If
```

File: docs/install/pubsub-service-account.md [hunk 9]
```diff
@@ -6,137 +6,137 @@ data plane component, which basically needs permissions to read and/or write to
...
+    there is no source instance using a Kubernetes Service Account, this
+    Kubernetes Service Account will be deleted, and its binding to the
+    Google Cloud Service Account will be deleted as well.
```
Comment (Harwayne) [inline docs/install/pubsub-service-account.md:118]:
```suggestion
    all sources using a Kubernetes Service Account are deleted then that Kubernetes Service Account, and its corresponding Google Cloud Service account, will be deleted.

File: docs/install/install-knative-gcp.md (hunk 2 above)
Comment (liu-cong) [reply docs/install/install-knative-gcp.md:64]: Added a TODO and created an issue: https://github.com/google/knative-gcp/issues/896

File: docs/install/authentication-machanisms-gcp.md [hunk 10]
```diff
@@ -0,0 +1,100 @@
...
+
+1. Enable Workload Identity.
+
+    ```shell
+    gcloud services enable iamcredentials.googleapis.com
+    gcloud beta container clusters update $CLUSTER_NAME \
+    --identity-namespace=$PROJECT_ID.svc.id.goog
+    ```
+
+2. Bind the Kubernetes Service Account `controller` with Google Cloud
```
Comment (yolocs) [inline docs/install/authentication-machanisms-gcp.md:58]: I think recommended way is always use `1.` Similar for other places.

File: docs/install/install-knative-gcp.md (hunk 1 above)
Comment (liu-cong) [reply docs/install/install-knative-gcp.md:29]: Good call. Done.

File: docs/install/pubsub-service-account.md [hunk 11]
```diff
@@ -6,137 +6,137 @@ data plane component, which basically needs permissions to read and/or write to
...
+In general, we would just need permissions to
+receive messages (`roles/pubsub.subscriber`). However, in the case of the
+`Channel`, we would also need the ability to publish messages
+(`roles/pubsub.publisher`).
 
-        ```shell
-        gcloud iam service-accounts create cre-pubsub
-        ```
+1. Create a new Service Account named `cre-pubsub` with the following
+    =command:
```
Comment (yolocs) [inline docs/install/pubsub-service-account.md:33]: What's the `=` about?

**Comment by liu-cong** [comment]
Thanks @Harwayne for the detailed review!

File: docs/install/pubsub-service-account.md (hunk 11 above)
Comment (liu-cong) [reply docs/install/pubsub-service-account.md:33]: removed

File: docs/install/authentication-machanisms-gcp.md (hunk 10 above)
Comment (liu-cong) [reply docs/install/authentication-machanisms-gcp.md:58]: Done.
